BSP/Inc/BSP_I2C.h

```diff
@@ -0,0 +1,63 @@
+/* Copyright (c) 2018-2023 UT Longhorn Racing Solar */
+// BSP I2C header file
+
+#ifndef BSP_I2C_H
+#define BSP_I2C_H
+
+#include <stdint.h>
+#include "BSP.h"
+#include "stm32f4xx_hal.h"
+#include "stm32f4xx_hal_i2c.h"
+#include "stm32f4xx_hal_rcc.h"
+#include "stm32f4xx_hal_rcc_ex.h"
+#include "stm32f4xx_hal_gpio.h"
+
+/**
+ * @brief Initialize I2C port for EEPROM interaction
+ * 
+ */
+BSP_Status BSP_I2C_Init();
+
+/**
+* @brief   Transmits data onto the I2C bus.
```

Comment:
Typically I don't use the colons on doxygen style comments. e.g.:
@param hi2c struct with config data

because I think the colons are not interpreted super well by VSCode and just putting the name right before the description seems to work. up to you though it doesn't really matter.